scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -383,6 +383,10 @@ def generate_command(self):
             '--log-cli-format=%(levelname)s: %(message)s'
         ])
 
+        # Use the test timeout as the base timeout for pytest
+        base_timeout = handler.get_test_timeout()
```

Comment:
I'm concerned about using this timeout as base timeout in pytest-harness. This timeout (from Twister config file) is used here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/pylib/twister/twisterlib/harness.py#L474
when Twister runs subprocess with pytest-harness to kill process in case of timeout expires.

When you use this timeout in pytest, then you can stuck for long time, when is a problem with flashing or with getting output from shell (in long tests, when timeout is set to longer time or when timeout multiplier is used this can be problematic).
In pytest-harness, couple of timeouts are used. When you look into shell fixture, there is given as default 20s:
https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/pylib/pytest-twister-harness/src/twister_harness/fixtures.py#L60
if you need longer value, then you can overwrite this fixture in your test dir to use proper configs.
For flashing there is used  another timeout ... so maybe you can look there to extend the default.

If you want to use `--base-timeout` here, to pass it from Twister to pytest-harness, consider to use it only for cases where it is needed, to use some predefined value for a platform (e.g. extended base timeout for qemu).

Reply:
from the other hand ... maybe we should try to keep it simple and to use Twister timeout as a base-timeout in pytest-harness, and just change (lower) some timeouts in pytest-harness implementation (as is for shell fixture, readlines  methods or for flashing).

Reply:
At first, I thought to limit this to when `--device-flash-with-test` is enabled, but then this would be inconsistent with testing without that flag.

On the other hand, this would behave similar to the console harness where we would wait up to timeout value if there is no output.

Reply:
Ok, as I wrote, keep it simple, do not add additional timeouts to command line interface. We can use `--base-timeout`  and if user want to change samething, then can overwrite `dut` or `shell` fixture, to change any specific timeouts.